src/handlers/indexers/create_indexer.rs

```diff
@@ -14,57 +15,77 @@ use crate::infra::repositories::indexer_repository::{self, IndexerDb};
 use crate::publishers::indexers::publish_start_indexer;
 use crate::AppState;
 
+#[derive(Default)]
+struct CreateIndexerRequest {
+    webhook_url: String,
+    data: Bytes,
+}
+
+// not using From trait as we need async functions
+async fn build_create_indexer_request(request: &mut Multipart) -> Result<CreateIndexerRequest, IndexerError> {
+    let mut create_indexer_request = CreateIndexerRequest::default();
+    while let Some(field) = request.next_field().await.map_err(IndexerError::FailedToReadMultipartField)? {
+        let field_name = field.name().ok_or(IndexerError::InternalServerError)?;
+        match field_name {
+            "script.js" => {
+                create_indexer_request.data = field.bytes().await.map_err(IndexerError::FailedToReadMultipartField)?
+            }
+            "webhook_url" => {
+                create_indexer_request.webhook_url =
+                    field.text().await.map_err(IndexerError::FailedToReadMultipartField)?
+            }
+            &_ => return Err(IndexerError::UnexpectedMultipartField(field_name.to_string())),
```

Comment:
is the `&` a typo?

Reply:
ya, fixed, but it still worked apparently